libvirt/domain.go

```diff
@@ -216,6 +216,20 @@ func newDiskForCloudInit(virConn *libvirt.Connect, volumeKey string) (libvirtxml
 			Type: "raw",
 		},
 	}
+	if runtime.GOARCH == "s390x" || runtime.GOARCH == "s390" {
```

Comment:
I have a simple test on x86, it could create a VM successfully. But I will make more test on x86 and s390 when I finish the code.

Reply:
@jaypoulz Should we support this scenario? We have to make clear of this at first.

Reply:
If we have to support this, can we get remote arch by calling libvirt API?

Reply:
I will change my code, and try to get remote arch by calling libvirt API.

Reply:
@xkwangcn this was brought up today in a meeting - we are using an x86_64 host as the testing node, so it will be important for this to pull in the remote arch for this logic.

Reply:
@jaypoulz yes, I have tested: 
terraform apply on s390; remote libvirt on x86;
terraform apply on s390; remote libvirt on s390; 
the above two cases could work success. 
I will test more cases, for example, 
terraform apply on x86; remote libvirt on s390;
terraform apply on x86; remote libvirt on x86; 
and the local test for both x86 and s390.

But now the CI build failed, I don't know the reason. 
Next I will modify my code to pass CI build.